lib/sendf.c

```diff
@@ -616,8 +616,6 @@ CURLcode Curl_client_write(struct Curl_easy *data,
                            size_t len)
 {
   struct connectdata *conn = data->conn;
-  if(0 == len)
-    len = strlen(ptr);
```

Comment:
how about also adding...
~~~c
DEBUGASSERT(len);
~~~
... to better catch mistakes at least in debug-builds.